pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -4155,6 +4155,257 @@ def prepare_plots(self):
                 }
 
 
+class FluxlineCrosstalkAnalysis(MultiQubit_TimeDomain_Analysis):
+    def __init__(self, qb_names, *args, **kwargs):
+        params_dict = {f'{qbn}.amp_to_freq_model':
+                       f'Instrument settings.{qbn}.fit_ge_freq_from_flux_pulse_amp'
+                       for qbn in qb_names}
+        kwargs['params_dict'] = kwargs.get('params_dict', {})
+        kwargs['params_dict'].update(params_dict)
+        super().__init__(qb_names, *args, **kwargs)
+
+    def process_data(self):
+        super().process_data()
+        if self.sp is None:
+            raise ValueError('This analysis needs a SweepPoints '
+                             'class instance.')
+
+        pdd = self.proc_data_dict
+        # get the ramsey phases as the values of the first sweep parameter
+        # in the 1st sweep dimension.
+        # !!! This assumes all qubits have the same ramsey phases !!!
+        pdd['ramsey_phases'] = self.sp.get_sweep_params_property('values', 0)
```

Comment:
This is also done in the DriveCrosstalkCancellationAnalysis, form where this part of the code was copied, I assume.
Ants wrote both classes. I added this warning comment when I went through the DriveCrosstalkCancellationAnalysis code and did minor improvements like a using the self.sp variable. So I think @antsr should answer the question whether we should relax this assumption. I can work on making it general if we want this.

Reply:
I don't think we'll ever want to sweep the Ramsey phases over different values for different qubits. It could be implemented for this measurement in principle, but I don't see the point now. If this is moved to the quantum experiment framework at some point in the future, we would get that for free.